Fix RPC failure testing (continuation of #9707)

jnewbery opened the pull request:
Commit 9db8eecac1c713c760c0217b6acb7455c657fa8b improved the
assert_raises_jsonrpc() function for better testing of RPC failure
modes. This commit continues the job by removing remaining broken
try-except RPC testing from individual test cases and replacing it
with calls to assert_raises_jsonrpc().

Note that this does not fully complete the job. This commit replaces the try/except code with assert_raises_jsonrpc() only for those test cases where no changes to bitcoind were required (ie the test cases weren't necessarily fully testing the behaviour, but the behaviour was correct anyway). For any test cases where fixing the test code has revealed bugs or inconsistencies, I'll raise a separate issue/PR (eg #9738, #9714, #9841, #9710). There are a few more of those to come.

jonasschnelli commented:
Thanks for the cleanup. Factor out the try/catches makes the tests much more readable.
utACK f59cacd0e0e210783bc7eace5e317a7b466f719b

jtimon commented on qa/rpc-tests/blockchain.py:
```diff
@@ -11,14 +11,13 @@
     assert_is_hash_string,
     start_nodes,
-    connect_nodes_bi,
+    connect_nodes_bi
```
I don't think you need to remove the coma here.

jnewbery committed:
Fix RPC failure testing (2 of 2)

Commit 9db8eecac1c713c760c0217b6acb7455c657fa8b improved the
assert_raises_jsonrpc() function for better testing of RPC failure
modes. This commit completes the job by removing remaining broken
try-except RPC testing from the individual test cases and replacing it
with calls to assert_raises_jsonrpc().

jnewbery commented:
rebased and reverted the comma removal: https://github.com/jnewbery/bitcoin/commit/c3d6c83d67f2b50b17c93ffb74559420a7cfb9ba / https://github.com/jnewbery/bitcoin/tree/pr9842.2

jnewbery commented:
Bad rebase. Rebased to https://github.com/jnewbery/bitcoin/commit/4b7749550657ace31e91f56fc336ced532605f74 / https://github.com/jnewbery/bitcoin/tree/pr9842.3

jnewbery commented:
oops. Merge issues from priority removal. Should be fixed now.

jnewbery commented:
rebased

jnewbery commented:
rebased. Any chance of getting this merged soon?

MarcoFalke commented:
utACK c9bd0f64212cd600daf30c9c59035bfe9f07b81b
